### PR TITLE
SAK-46615 GradebookNG: Importing grades by spreadsheets is not workin…

### PR DESCRIPTION
**Pull request by josecebe**
…g when items contains comma as decimal separator

### CODE REVIEW

**Review by bjones86 — DISMISSED** [review]

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/importExport/GradeImportConfirmationStep.java [hunk 1]
```diff
@@ -203,7 +203,8 @@ public void onSubmit(AjaxRequestTarget target, Form<?> form) {
 				// Modify any that need modification
 				itemsToModify.forEach(item -> {
 
-					final Double points = NumberUtils.toDouble(item.getItemPointValue());
+					final String itemPoints = StringUtils.replace(item.getItemPointValue(), ",", ".");
+					final Double points = NumberUtils.toDouble(itemPoints);
```
Comment (ottenhoff) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/importExport/GradeImportConfirmationStep.java:207]: replace commas with periods just seems so wrong ..... no?!

why don't we have a centralized method to do this by now? why aren't there unit tests?

Comment (bjones86) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/importExport/GradeImportConfirmationStep.java:207]: Unfortunately this happens in many places in GBNG

Comment (ottenhoff) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/importExport/GradeImportConfirmationStep.java:207]: you're okay with your instructor commas being replaced with periods?!

  private static String formatGradeForLocale(final String grade, final Locale locale) {
    if (StringUtils.isBlank(grade)) {
      return "";
    }

    String s;
    try {
      final DecimalFormat df = (DecimalFormat) NumberFormat.getInstance(locale);
      final Double d = df.parse(grade).doubleValue();

      df.setMinimumFractionDigits(0);
      df.setGroupingUsed(false);

      s = df.format(d);
    } catch (final NumberFormatException | ParseException e) {
      log.warn("Bad format, returning original string: {}", grade);
      s = grade;
    }

    return StringUtils.removeEnd(s, ".0");
  }

Comment (bjones86) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/importExport/GradeImportConfirmationStep.java:207]: This is in the context of a grade cell in a spreadsheet. Previously in a decimal locale, if a comma was present this code would explode. So, yes, this is arguable, but marginally better than before.

If you would rather see a completely new centralized approach with unit tests, I'm fine with that as well. I just didn't think this was the JIRA to do that for.

Comment (josecebe) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/importExport/GradeImportConfirmationStep.java:207]: I agree that this is probably not the best way to do it, but I found several places where this replacement method is used to do the same. We should probably work on a centralized method to do this.

Comment (ottenhoff) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/importExport/GradeImportConfirmationStep.java:207]: Sorry but this just introduces a new bug for default (USA) locale!

1. Create a new GB item worth 1000 points
2. Export to CSV
3. Modify CSV by hand
4. Note the header and the comma! "Student ID","Name","# Student Number","New Item [1,000]"," * New Item"
5. Modify one character and make it worth 1,001 points
6. Import

Comment (ottenhoff) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/importExport/GradeImportConfirmationStep.java:207]: gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/util/FormatHelper.java seems like the right place.... or you could go to kernel .... knowing the locale seems important before doing a replace

There are tests here:

gradebookng/tool/src/test/org/sakaiproject/gradebookng/business/TestGradebookNgBusinessService.java

Comment (josecebe) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/importExport/GradeImportConfirmationStep.java]: Hi again, I have fixed the PR using the FormatHelper class

**Review by ottenhoff — APPROVED** [review]

**Review by mpellicer — APPROVED** [review]

**Comment by mpellicer** [comment]
Awesome @josecebe !

**Pull request merged**